### DISCUSSION TITLE
Edge probe

### DISCUSSION THREAD

**Discussion by jratha** [winder/Universal-G-Code-Sender#2283] [Help]
Hi all

I’ve got a little conundrum. I’m trying to set a probe to set the my work position correctly on my Arduino gbrl cnc.

```
G38.2 Z-40 F60;
G92 Z5.80;
G91;
G0 Z5;
M30

G38.2 X-50 F60;
G92 X3.175;
G91;
G0 X10;
M30

G38.2 Y50 F60;
G92 Y3.175;
G91;
G0 Y10;
M30
```

These are the macros I’ve set up in universal GCode sender, both the y axis and z axis work perfectly, but when I initiate the x axis, it attempts to move but instead instantly executes the G0 X10 retract command.

Has anyone else had this issue or is it just me. Can anyone help me out with this?

**Comment by winder** [comment]
What version are you running?

I think some people have had success adding dwell commands (i.e. `G4 P0.5`) after probe macros to make things more consistent.

There is also a probe plugin in the platform which might be useful.

**Comment by jratha** [comment]
Hi will, I’m using version 2 platform, but the net beans version rather than the nightly build, I was waiting for something a bit more stable as I have problems with the nightly’s on my old laptop 

I have seen the probe plug-in but I’m using inside rather than outside measurements 

I will give the dwell command a try tomorrow when I get chance

**Comment by jratha** [comment]
Im a bit of a noob at this and just getting into gcode programming, but im not sure where i sould put the dwell command in the macro. should it look like this?

`G38.2 X-50 F60;
G4 P0.5;
G92 X3.175;
G91;
G0 X10;
M30`

**Comment by AbySet** [comment]
Just a bit gess, can you check $2, $3 and $10 values in grbl settings ?

**Comment by jratha** [comment]
$2 is set to 0, as is $3. $10 is set to 16

**Comment by AbySet** [comment]
Sorry, my bad, i meant $20 not $10.

**Comment by jratha** [comment]
$20=0. I have checked on both my uno board and my mega 2560. Both are set the same, when i use the mega, all works fine but when I use the uno, the touch probe moves 0.20mm and the retracts

**Comment by AbySet** [comment]
Sorry i'm late, i was busy, ok i'll put your macros in my machine and test them out, i have arduino uno, will let you know the result.

**Comment by AbySet** [comment]
Ok i have tested your macros and they are working just fine for me.

My specs:
Windows 10 x64
grbl 1.1
ugsplatform nighly build (october,7,2018)
arduino uno

At this point i would reflash my arduino with grbl 1.1.
Just one question: Is wiring ok to the X axis steeper motor ? check it into the arduino uno plugs (just in case).

**Comment by jratha** [comment]
Yep, steppers for x all wired up correctly, will try the new nightly build in the morning and will try reflashing my uno and let you know asap

**Comment by jratha** [comment]
Well played about with this for last couple of days and still not got it to work, rewired all the pins to my uno pins  but still the same problem.

The only difference is I’m using a fresh install of windows 7. 

I’ve tried the ugs with my mega 2560 board and everything works fine, so I’m presuming that this could be a fault with the uno board?

**Comment by carneeki** [comment]
I've had an issue with one board where I might have shorted something and
fried most of the outputs of the microcontroller - it appeared dead, yet
the serial interface was still good. The only way I realised what had
happened was when I wrote a small Arduino sketch to turn on all pins, and
used a multimeter to check the voltage, which from memory was about 0.3V
when on and 0V when off.

If you're willing to try it, disconnect the arduino from your steppers, and
run a multimeter or LED and resistor from ground up to you pins being
tested, and have them blink on and off. I'd imagine that if one is out, all
of them from the PORT will be out. If so, the board might be toast (or more
likely the uC, you _can_ desolder and replace it, but the boards are cheap
enough that it usually isn't worth the effort).

Good luck!

On Sun, 21 Oct 2018 at 05:46, jratha <notifications@github.com> wrote:

> Well played about with this for last couple of days and still not got it
> to work, rewired all the pins to my uno pins but still the same problem.
>
> The only difference is I’m using a fresh install of windows 7.
>
> I’ve tried the ugs with my mega 2560 board and everything works fine, so
> I’m presuming that this could be a fault with the uno board?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/winder/Universal-G-Code-Sender/issues/1139#issuecomment-431608168>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAdYqGWLTxEw-POss3Aq2LucZ4dYsoKRks5um29ngaJpZM4XX6PX>
> .
>


-- 
Adam "carneeki" Carmichael
p: +61 415 37 1990
w: http://bigneek.com
e: carneeki@carneeki.net
i: 2207644

**Comment by AbySet** [comment]
Can you tell us what kind of motors drivers you are using ?
Try to switch drivers between Y and X :+1: 
If the issue still on X axis then your arduino or it's flash may be bad.
If the issue skip to Y axis then the driver on Y (whitch was on X before) is bad.
But i dont think it's a driver issue cz you tryed it with mega 2560 and it was fine.
Did you reflash arduino uno ?

**Comment by jratha** [comment]
I’m using tb6560 drivers. Had some spare and have tried swapping already

**Comment by AbySet** [comment]
Sorry i'm late, ok then your drivers are ok.
Did you reflash grbl into arduino uno ?
Are you using original arduino or cloned one ? some people are reporting bugs with coloned ones.
